src/Bicep.Core/TypeSystem/ResourceType.cs

```diff
@@ -4,18 +4,25 @@
 
 namespace Bicep.Core.TypeSystem
 {
+    public record ResourceTypeComponents(
+        ResourceTypeReference TypeReference,
+        ResourceScope ValidParentScopes,
+        ITypeReference Body);
+
     public class ResourceType : TypeSymbol, IScopeReference
     {
-        public ResourceType(ResourceTypeReference typeReference, ResourceScope validParentScopes, ITypeReference body)
+        public ResourceType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceScope validParentScopes, ITypeReference body)
```

Comment:
This is really similar to what I ended up doing in my fork. What's on `NamespaceType`? Does that capture all the info about the provider?

Reply:
Yeah - `NamespaceType` should capture everything about an imported or built-in namespace. I've made a few changes here recently so you may want to sync your fork with `main` to pick them up.